crates/miden-objects/src/note/storage.rs

```diff
@@ -0,0 +1,190 @@
+use alloc::vec::Vec;
+
+use crate::errors::NoteError;
+use crate::utils::serde::{
+    ByteReader,
+    ByteWriter,
+    Deserializable,
+    DeserializationError,
+    Serializable,
+};
+use crate::{Felt, Hasher, MAX_STORAGE_VALUE_PER_NOTE, WORD_SIZE, Word, ZERO};
+
+// NOTE STORAGE
+// ================================================================================================
+
+/// A container for note storage.
+///
+/// A note can be associated with up to 128 storage items. Each value is represented by a single
+/// field element. Thus, note storage items can contain up to ~1 KB of data.
+///
+/// All storage items associated with a note can be reduced to a single commitment which is
+/// computed by first padding the values with ZEROs to the next multiple of 8, and then by computing
+/// a sequential hash of the resulting elements.
+#[derive(Clone, Debug)]
+pub struct NoteStorage {
+    items: Vec<Felt>,
+    commitment: Word,
+}
+
+impl NoteStorage {
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns [NoteStorage] instantiated from the provided values.
+    ///
+    /// # Errors
+    /// Returns an error if the number of provided storage is greater than 128.
+    pub fn new(values: Vec<Felt>) -> Result<Self, NoteError> {
```

Comment:
In this file, I'd also rename `value` to `item` so it matches the `items` field.